addons/stock_account/report/report_stock_forecasted.py

```diff
@@ -12,9 +12,16 @@ def _compute_draft_quantity_count(self, product_template_ids, product_variant_id
         """ Overrides to computes the valuations of the stock. """
         res = super()._compute_draft_quantity_count(product_template_ids, product_variant_ids, wh_location_ids)
         domain = self._product_domain(product_template_ids, product_variant_ids)
-        svl = self.env['stock.valuation.layer'].search(domain)
+        wh_company = self.env['stock.location'].search([('id', '=', wh_location_ids)], limit=1).company_id
+        svl = self.env['stock.valuation.layer'].search(domain + [('company_id', '=', wh_company.id)])
         currency = svl.currency_id or self.env.company.currency_id
-        value = float_repr(sum(svl.mapped('value')), precision_digits=currency.decimal_places)
+        quantity = sum(svl.filtered(lambda layer: layer.stock_move_id.location_dest_id.id in wh_location_ids).mapped('quantity'))
```

Comment:
the filter can't be in the search?

Reply:
I don't think so. Because the `svl` are used without filters here: https://github.com/odoo/odoo/pull/64942/files#diff-f5d2c5e4790ab54c4f6b7f69f0f46356a9e26a735c1c53fa2446f3c6463546d1R20